tests/test_formats.py

```diff
@@ -66,6 +66,24 @@ def test_datetimeformat(dt, formatted_datetime):
     assert datetimeformat(dt) == formatted_datetime
 
 
+@pytest.mark.parametrize("dt, formatted_datetime", (
+    (datetime(2012, 11, 10, 9, 8, 7, 6), "9am GMT, Saturday 10 November 2012"),
+    ("2012-11-10T09:08:07.0Z", "9am GMT, Saturday 10 November 2012"),
+    (datetime(2012, 8, 10, 9, 8, 7, 6), "10am BST, Friday 10 August 2012"),
+    ("2012-08-10T09:08:07.0Z", "10am BST, Friday 10 August 2012"),
+    (datetime(2012, 8, 10, 9, 8, 7, 6, tzinfo=pytz.utc), "10am BST, Friday 10 August 2012"),
+    (datetime(2012, 8, 1, 9, 8, 7, 6, tzinfo=pytz.utc), "10am BST, Wednesday 1 August 2012"),
+    (datetime(2012, 8, 1, 22, 59, 7, 6, tzinfo=pytz.utc), "11pm BST, Wednesday 1 August 2012"),
+    # Daylight savings edge case
+    (datetime(2012, 3, 25, 0, 59, 7, 6, tzinfo=pytz.utc), "12am GMT, Sunday 25 March 2012"),
+    (datetime(2012, 3, 25, 1, 59, 7, 6, tzinfo=pytz.utc), "2am BST, Sunday 25 March 2012"),
+    # Fall back to default if no valid date supplied
+    (None, None),
+))
+def test_utctoshorttimelongdateformat(dt, formatted_datetime):
```

Comment:
Worth testing that the `default_value` and `localize` parameters work as expected when not left as default values?